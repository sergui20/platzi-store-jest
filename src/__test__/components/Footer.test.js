import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer'; // This package will render React components to pure JS objects, without the need of the DOM
import Footer from '../../components/Footer';

describe('<Footer> </Footer>', () => {
    const footer = mount(<Footer />);

    test('Should render Footer component', () =>  {
        expect(footer.length).toEqual(1)
    });

    test('Should render the title', () => {
        expect(footer.find(".Footer-title").text()).toBe('Platzi Store')
    });
});

// Use Snapshots when you want to test code that doesn't change drastically.
// In this use case, the Footer is a component that won't change.
describe('Footer Snapshot', () => {
    test('Check Footer UI', () => {
        const footer = create(<Footer />);
        expect(footer.toJSON()).toMatchSnapshot();
    });
});
