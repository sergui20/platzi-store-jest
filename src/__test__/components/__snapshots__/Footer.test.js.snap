// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot Check Footer UI 1`] = `
<div
  className="Footer updated-snapshot"
>
  <p
    className="Footer-title"
  >
    Platzi Store
  </p>
  <p
    className="Footer-copy"
  >
    Todos los Izquierdos Reservados
  </p>
</div>
`;
